### DISCUSSION TITLE
配列の宣言でのエラーについて

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#64] [Question-original]
１　エラーは出ていないのですがボタンでページ移動ができないので、その原因を見つけるために、最初に戻って、画像をボタンで何個も出すコードを修正してみました。
普通の画像ではできたのですが、配列にするとなぜかうまくいかないです。

２　画像ひとつひとつに変数を持ってもらいたいので、配列で宣言したいので、ボタンを押したときに生成される乱数によって、それぞれ画像を表示し、0から順番に配列に画像情報を入れているのですがうまくいかないです。

４　
@IBAction func Renda(){
        num=Int.random(in: 1...100)
        let xrand=Double.random(in: 40...370)
        let yrand=Double.random(in: 40...450)
        if num<=75{
            let image1:UIImage = UIImage(named:"Stone")!
            StoneImage[stonenum]=UIImageView(image:image1)
            StoneImage[stonenum].center = CGPoint(x:xrand, y:yrand)
            view.addSubview(StoneImage[stonenum])
            stonenum=stonenum+1
        }
        if (num>=76)&&(num<=95){
            let image1:UIImage = UIImage(named:"Stone")!
            EnemyImage[enemynum]=UIImageView(image:image1)
            EnemyImage[enemynum].center = CGPoint(x:xrand, y:yrand)
            view.addSubview(EnemyImage[enemynum])
            enemynum=enemynum+1
        }
        if num>=96{
            let image1:UIImage = UIImage(named:"Timer")!
            let TimerImage=UIImageView(image:image1)
            TimerImage.center = CGPoint(x:xrand, y:yrand)
            view.addSubview(TimerImage)
         }
    }

５　https://github.com/ShunHigu/Original

**Comment by Yonpachi48** [comment]
画面遷移をさせたいということであってるかな？
どの挙動をさせたら画面遷移をさせたいのかを明記してほしいです！
また、そこに該当するコードも見せてほしいです！

**Reply by ShunHigu** [reply]
スタート画面からボタンを押して次の画面に有働できないというエラーでした。
ボタンをタップ後に画面が落ち、AppDelegateの@mainに次のエラー分がありました。
Thread 1: "[<Original.Stage1ViewController 0x7ff790306f00> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key EnemyImage."

原因は定められていないImageViewがあったことでした。
